Packages/com.unity.inputsystem/InputSystem/Actions/InputBindingResolver.cs

```diff
@@ -252,6 +252,74 @@ public unsafe void AddActionMap(InputActionMap map)
+                            {
+                                // Search in devices for only this map.
+                                var list = devicesForThisMap.Value;
+                                for (var i = 0; i < list.Count; ++i)
+                                {
+                                    var device = list[i];
+                                    if (!device.added)
+                                        continue; // Skip devices that have been removed.
+                                    numControls += InputControlPath.TryFindControls(device, path, 0, ref resolvedControls);
+                                }
+                            }
+                            else
+                            {
+                                // Search globally.
+                                numControls = InputSystem.FindControls(path, ref resolvedControls);
+                            }
+
+                            // Check for controls that are already bound to the action through other
+                            // bindings. The first binding that grabs a specific control gets to "own" it.
+                            if (numControls > 0)
+                            {
+                                for (var i = 0; i < n; ++i)
+                                {
+                                    ref var otherBindingState = ref bindingStatesPtr[bindingStartIndex + i];
+
+                                    // Skip if binding has no controls.
+                                    if (otherBindingState.controlCount == 0)
+                                        continue;
+
+                                    // Skip if binding isn't from same action.
+                                    if (otherBindingState.actionIndex != actionStartIndex + actionIndexInMap)
+                                        continue;
+
+                                    // Check for controls in the set that we just resolved that are also on the other
+                                    // binding. Each such control we find, we kick out of the list.
+                                    for (var k = 0; k < numControls; ++k)
+                                    {
+                                        var controlOnCurrentBinding = resolvedControls[firstControlIndex + k - memory.controlCount];
+                                        var controlIndexOnOtherBinding = resolvedControls.IndexOf(controlOnCurrentBinding,
+                                            otherBindingState.controlStartIndex - memory.controlCount, otherBindingState.controlCount);
+                                        if (controlIndexOnOtherBinding != -1)
+                                        {
+                                            // Control is bound to a previous binding. Remove it from the current binding.
+                                            resolvedControls.RemoveAt(firstControlIndex + k - memory.controlCount);
```

Comment:
A bit sad that this needs to remove bindings. A more efficient way would be to pass a list of "ignored controls" to `FindControls` imho.

Reply:
Good point. Looked into it and I think adding an argument for a collection of ignored controls would be a straightforward change. Unfortunately, it's only *some* controls in `resolvedControls` that should be ignored and they don't have to be one after the other in the list. That part seems like it'd make moving the check into `FindControls` quite messy.